Mage.Sets/src/mage/cards/m/MiragePhalanx.java

```diff
@@ -0,0 +1,108 @@
+                    "except it has haste and loses soulbond. " +
+                    "Exile it at end of combat.\"";
+
+    public MiragePhalanx(UUID ownderId, CardSetInfo setInfo) {
+        super(ownderId, setInfo, new CardType[]{CardType.CREATURE}, "{4}{R}{R}");
+
+        this.addSubType(SubType.HUMAN);
+        this.addSubType(SubType.SOLDIER);
+        this.power = new MageInt(4);
+        this.toughness = new MageInt(4);
+
+        // Soulbond
+        this.addAbility(new SoulbondAbility());
+
+        // As long as Mirage Phalanx is paired with another creature, each of those creatures has
+        // “At the beginning of combat on your turn, create a token that’s a copy of this creature,
+        // except it has haste and loses soulbond.
+        // Exile it at end of combat.”
+        Ability ability = new BeginningOfCombatTriggeredAbility(new MiragePhalanxEffect(), TargetController.YOU, false);
+        this.addAbility(new SimpleStaticAbility(Zone.BATTLEFIELD, new GainAbilityPairedEffect(ability, ruleText)));
+    }
+
+    private MiragePhalanx(final MiragePhalanx card) {
+        super(card);
+    }
+
+    @Override
+    public MiragePhalanx copy() {
+        return new MiragePhalanx(this);
+    }
+}
+
+class MiragePhalanxEffect extends OneShotEffect {
+    MiragePhalanxEffect() {
+        super(Outcome.PutCreatureInPlay);
+        this.staticText = "create a token that’s a copy of this creature, " +
+                "except it has haste and loses soulbond. " +
+                "Exile it at end of combat.";
+    }
+
+    private MiragePhalanxEffect(final MiragePhalanxEffect effect) { super(effect); }
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+        Permanent permanent = game.getPermanent(source.getSourceId());
+        if (permanent == null) { return false; }
+
+        // Create the token, it has haste
+        CreateTokenCopyTargetEffect tokenCopyEffect = new CreateTokenCopyTargetEffect(source.getControllerId(), null, true);
+        tokenCopyEffect.setTargetPointer(new FixedTarget(permanent, game));
+        tokenCopyEffect.apply(game, source);
+        if (tokenCopyEffect.getAddedPermanent().isEmpty()) { return false; }
+
+        // Change abilities on the token
+        for (Permanent addedToken : tokenCopyEffect.getAddedPermanent()) {
+            // It loses soulbond
+            for (Ability ability : addedToken.getAbilities()) {
+                if (ability instanceof SoulbondAbility) {
+                    LoseAbilityTargetEffect loseSoulbondEffect = new LoseAbilityTargetEffect(ability);
+                    loseSoulbondEffect.setTargetPointer(new FixedTarget(addedToken.getId()));
+                    game.addEffect(loseSoulbondEffect, source);
+                }
+            }
+            // Exile it at the end of combat
+            ExileTargetEffect exileEffect = new ExileTargetEffect("Exile it at the end of combat.");
```

Comment:
this should be a single trigger that applies to all the tokens rather than individual ones

Reply:
How come it's a single trigger?

Reply:
it’s because of how the rules work involving token doublers, the game sets up a single trigger to apply to a single token but it applies to multiple tokens if there’s a doubler

Reply:
I am also not sure how I would exile all of them with a single trigger.

Reply:
no, those ones are incorrect as well and it’s something I’ve been meaning to fix

you should be able to exile them all with a single trigger by using `FixedTargets`

Reply:
I see. Well, once this card gets sorted out, I can add the same fix to this PR